src/modules/home/api/list-accommodations.api.ts

```diff
@@ -0,0 +1,29 @@
+import { useQuery } from '@tanstack/react-query';
+import { apiClient } from '#/shared/libs/api-client.lib';
+import { useSearchParams } from 'react-router-dom';
+import { type AccommodationListQueryParams } from '#/modules/home/types/accommodation-list-query-params.type';
+import { type AccommodationListResponse } from '../types/accommodation-list.res.type';
+
+export function useListAccommodationsQuery(page: number, limit: number) {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  function setQueryParams(queryParams: AccommodationListQueryParams) {
```

Comment:
Move it to the page or additional hook level (useListAccomodationSearchParams) + add validation there